flex-templates/java/regional_dlp_de_identification/src/main/java/org/apache/beam/samples/DLPTextToBigQueryStreaming.java

```diff
@@ -0,0 +1,780 @@
+    }
+
+    @FinishBundle
+    public void finishBundle() throws Exception {
+      if (this.dlpServiceClient != null) {
+        this.dlpServiceClient.close();
+      }
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      String key = c.element().getKey();
+      Table nonEncryptedData = c.element().getValue();
+      ContentItem tableItem = ContentItem.newBuilder().setTable(nonEncryptedData).build();
+      this.requestBuilder.setItem(tableItem);
+      DeidentifyContentResponse response = dlpServiceClient.deidentifyContent(this.requestBuilder.build());
+      Table tokenizedData = response.getItem().getTable();
+      numberOfRowsTokenized.update(tokenizedData.getRowsList().size());
+      numberOfBytesTokenized.update(tokenizedData.toByteArray().length);
+      c.output(KV.of(key, tokenizedData));
+    }
+  }
+
+  /**
+   * The {@link TableRowProcessorDoFn} class process tokenized DLP tables and
+   * convert them to
+   * BigQuery Table Row.
+   */
+  public static class TableRowProcessorDoFn extends DoFn<KV<String, Table>, KV<String, TableRow>> {
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      Table tokenizedData = c.element().getValue();
+      List<String> headers = tokenizedData.getHeadersList().stream()
+          .map(fid -> fid.getName())
+          .collect(Collectors.toList());
+      List<Table.Row> outputRows = tokenizedData.getRowsList();
+      if (outputRows.size() > 0) {
+        for (Table.Row outputRow : outputRows) {
+          if (outputRow.getValuesCount() != headers.size()) {
+            throw new IllegalArgumentException(
+                "CSV file's header count must exactly match with data element count");
+          }
+          c.output(
+              KV.of(
+                  c.element().getKey(),
+                  createBqRow(outputRow, headers.toArray(new String[headers.size()]))));
+        }
+      }
+    }
+
+    private static TableRow createBqRow(Table.Row tokenizedValue, String[] headers) {
+      TableRow bqRow = new TableRow();
+      AtomicInteger headerIndex = new AtomicInteger(0);
+      tokenizedValue
+          .getValuesList()
+          .forEach(
+              value -> {
+                String checkedHeaderName = checkHeaderName(headers[headerIndex.getAndIncrement()].toString());
+                bqRow.set(checkedHeaderName, value.getStringValue());
+              });
+      return bqRow;
+    }
+  }
+
+  private static String getJsonSchema(ValueProvider<String> inputSchema) {
+    String []fields = inputSchema.get().split(",");
+    String jsonSchema = "{ \"fields\": [";
+
+    for (int i = 0; i < fields.length; i++) {
+      jsonSchema += String.format("{\"name\": \"%s\", \"type\": \"%s\"}%s", fields[i].split(":")[0].trim(), fields[i].split(":")[1].trim(),
+          i == fields.length - 1 ? "": ",");
+    }
+    jsonSchema += "]}";
+    return jsonSchema;
+  }
+
+  private static String getFileName(ReadableFile file) {
```

Comment:
consider renaming to something like validating file attirbutes, because this function does more than just get a file name, it seems.  Also, consider using the Path object, cuz I think it efficiently handles some of these things.

Reply:
@erlanderlo 
 Good review, I changed the class name as you suggested.
 Regarding the sentence `Also, consider using the Path object, cuz I think it efficiently handles some of these things.`
Currently in step 1 of the pipeline, the files are read from a bucket and passed to the now renamed "getAndValidateFileAttributes" function, which currently receives the complete file with all attributes. What you would be proposing is not to pass the whole file, but only its path, which would be `file.getMetadata().resourceId()`? What would then be passed to the function would be: `gs://BUCKET_NAME/FILE_NAME.csv`.